clash_lib/src/proxy/tuic/handle_task.rs

```diff
@@ -0,0 +1,155 @@
+                            "[udp] [{assoc_id:#06x}] [to-native] to {addr_display}: {err}"
+                        );
+                        Err(anyhow!(err))
+                    }
+                }
+            }
+            UdpRelayMode::Quic => {
+                tracing::info!("[udp] [{assoc_id:#06x}] [to-quic] {addr_display}");
+                match self.inner.packet_quic(pkt, addr, assoc_id).await {
+                    Ok(()) => Ok(()),
+                    Err(err) => {
+                        tracing::warn!(
+                            "[udp] [{assoc_id:#06x}] [to-quic] to {addr_display}: {err}"
+                        );
+                        Err(anyhow!(err))
+                    }
+                }
+            }
+        }
+    }
+
+    pub async fn dissociate(&self, assoc_id: u16) -> anyhow::Result<()> {
+        tracing::info!("[udp] [dissociate] [{assoc_id:#06x}]");
+        match self.inner.dissociate(assoc_id).await {
+            Ok(()) => Ok(()),
+            Err(err) => {
+                tracing::warn!("[udp] [dissociate] [{assoc_id:#06x}] {err}");
+                Err(anyhow!(err))
+            }
+        }
+    }
+
+    pub async fn heartbeat(self, heartbeat: Duration) {
+        loop {
+            time::sleep(heartbeat).await;
+
+            if self.is_closed() {
+                break;
+            }
+
+            if self.inner.task_connect_count() + self.inner.task_associate_count() == 0 {
+                continue;
+            }
+
+            match self.inner.heartbeat().await {
+                Ok(()) => tracing::trace!("[heartbeat]"),
+                Err(err) => tracing::error!("[heartbeat] {err}"),
+            }
+        }
+    }
+
+    pub async fn incoming_udp(&self, pkt: Packet) {
+        let assoc_id = pkt.assoc_id();
+        let pkt_id = pkt.pkt_id();
+
+        let mode = if pkt.is_from_native() {
+            "native"
+        } else if pkt.is_from_quic() {
+            "quic"
+        } else {
+            unreachable!()
+        };
+
+        tracing::info!(
+            "[udp] [{assoc_id:#06x}] [from-{mode}] [{pkt_id:#06x}] fragment {frag_id}/{frag_total}",
+            frag_id = pkt.frag_id() + 1,
+            frag_total = pkt.frag_total(),
+        );
+        match pkt.accept().await {
+            Ok(Some((data, remote_addr, _))) => {
+                tracing::info!("[udp] [{assoc_id:#06x}] [from-{mode}] [{pkt_id:#06x}] from {remote_addr}");
+                let (session, local_addr) = match self.udp_sessions.read().await.get(&assoc_id) {
+                    Some(v) => (v.incoming.clone(), v.local_addr.clone()),
+                    None => {
+                        tracing::error!("[udp] [{assoc_id:#06x}] [from-{mode}] [{pkt_id:#06x}] unable to find udp session");
+                        return;
+                    },
+                };
+                let remote_addr = match remote_addr {
+                    Address::None => unreachable!(),
```

Comment:
> Address type None is used in Packet commands that is not the first fragment of a UDP packet.

should we handle this case too?

Reply:
When UDP packet is not fully assembled, `pkt.accept().await` will return Ok(None).